woocommerce/woocommerce-ios
# [Woo POS] Adjust pagination height threshold. Fix `GhostItemCardView` rendering when no more items.

Part of https://github.com/woocommerce/woocommerce-ios/issues/14186


## Description
This PR addresses 2 items:
- Adds a threshold to infinite scrolling, so this is triggered when we've scrolled 70% through the current loaded view height
- Fixes an issue where the `GhostItemCardView` would always show at the end of the list when there are no more items to fetch from the remote.

https://github.com/user-attachments/assets/58608361-1289-4472-a1b8-d294bb294856


## Testing:
- For easiness, update the `ProductsRemote`'s `Constants.productsPerPage` to a lower threshold, like 7 or 10.
- Enter POS.
- Scroll down slowly, observe the right side of the list how as you scroll it re-positions itself as loads more items
- Scroll down fast, observe that if you reach the end of the currently-loaded item list, the temporary `GhostItemCardView` will appear while loading data
- Scroll to the end of the list, the `GhostItemCardView` may appear once, but no more after that.

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
